0.22.1 release branch

ogrisel opened the pull request:
Here is the rebase actions (updated):

```
d 947f37e3a MNT bump version to 0.23.dev0 and add new whats_new (#15631)
d b6f124c84 MAINT Improve variable order in BaseDecisionTree (#15664)
d 12196dab9 MNT replaced check_consistent_length, etc with _check_sample_weight in BaseGradientBoosting (#15478)
d 1c546cd9b CLN Move gradient and hessian closer to for loop in hist GBDT (#15686)
d a117db575 MNT Activates github actions (#15746)
d a91304ecd DOC Pre 0.22 release (#15735)
d d06ef03bd Configure twitter-action for @sklearn_commits (#15758)
pick cc88ec97b DOC fixed default values in dbscan (#15753)
d 479548ac9 MAINT remove placeholder github actions workflow (#15767)
pick c6f09ee5e DOC fix incorrect branch reference in contributing doc (#15779)
pick f344d3f24 DOC relabel Feature -> Efficiency in change log (#15770)
pick 0bbe923fe DOC fixed Birch default value (#15780)
d 69884d5fd ENH: Improves speed of one hot encoding (#15762)
d 3a3178ec3 MNT remove deprecated loss functions in gb.py (#15790)
d 4cdbde308 MNT remove deprecated ensemble.partial_dependence (#15789)
d 5573abbb9 MNT Removed deprecated logistic_regression_path (#15791)
pick ff6f88075 STY Minior change on code padding in website theme (#15768)
d 70ae89ecb MNT Removed deprecated metrics (#15794)
pick a83b8e0d4 DOC Fix yticklabels order in permutation importances example (#15799)
pick 5e66d8ee2 STY Update wrapper width (#15793)
pick 6f701e23e DOC Long sentence was hard to parse and ambiguous in _classification.py (#15769)
pick 3e26ea3cf DOC Removed duplicate 'classes_' attribute in Naive Bayes classifiers (#15811)
pick 1b1c869ef BUG Fixes pandas dataframe bug with boolean dtypes (#15797)
d ad80d3159 DEP Remove deprecated joblib tools in utils (#15792)
pick 4ce39dbc6 BUG Returns only public estimators in all_estimators (#15380)
pick e4c0adaf7 DOC improve doc for multiclass and types_of_target (#15333)
pick 0fa54e5d7 TST Increases tol for check_pca_float_dtype_preservation assertion (#15775)
d 9c62eee69 MNT Deprecate unused 'rotate' parameter in tree.plot_tree. (#15806)
d e94b67a4d ENH Allow two-element tuple as n_samples argument in make_circles and make_moons (#15707)
pick 9d61fe031 update _alpha_grid class in _coordinate_descent.py (#15835)
pick 29932e690 FIX Explicit conversion of ndarray to object dtype. (#15832)
d 1c42e79d4 FIX pass sample weights to final estimator (#15773)
pick 7dd12fb22 BLD Parallelize sphinx builds on circle ci (#15745)
d 4256542b1 MNT Removed deprecated attributes and parameters (#15803)
d 54da2f00f ENH do not allocate local arrays in Ridge*CV of store_cv_values is False (#15652)
pick 677f2c620 DOC correct url for preprocessing (#15853)
d 7ee0ae861 ENH CountVectorizer: sort features after pruning by frequency (#15834)
pick 5aa28afc4 MNT avoid generating too many cross links in examples (#15844)
pick 49b03befd DOC Correct wrong doc in precision_recall_fscore_support (#15833)
pick 62b1e266e DOC add comment in check_pca_float_dtype_preservation (#15819)
pick 64750e561 DOC correct indents in docstring _split.py (#15843)
pick d5c6c966c DOC fix docstring of KMeans based on sklearn guideline (#15754)
pick 893a4d455 DOC fix docstring of AgglomerativeClustering based on sklearn guideline (#15764)
pick 10eb17cd9 DOC fix docstring of AffinityPropagation based on sklearn guideline (#15777)
pick 64ac46370 DOC fixed SpectralCoclustering and SpectralBiclustering docstrings following sklearn guideline (#15778)
pick e504ea7dd DOC fix FeatureAgglomeration and MiniBatchKMeans docstring following sklearn guideline (#15809)
d a07974b3c FIX add best_score_ to Ridge*CV estimators (#15655)
d c7fccc697 DOC Correct best_score_ docstring in RidgeCV (#15859)
pick 6c582888a TST Specify random_state in test_cv_iterable_wrapper (#15829)
pick e0a8f7155 DOC Include LinearSV{C, R} in models that support sample_weights (#15871)
pick dc111489a DOC correct some indents (#15875)
d 42e17b348 MNT Removed deprecated attributes and parameters -- ctnd (#15804)
pick 41747f6ec DOC Fix documentation of default values in tree classes (#15870)
pick 6478e4bf6 DOC fix typo in docstring (#15887)
pick 43827c998 DOC FIX default value for xticks_rotation in plot_confusion_matrix (#15890)
d a05a5bc2b ENH Yield stack trace information in resilient mode model_selection warnings (#15622)
d 078e3ef29 [MRG] MNT Fix assert run python script (#15905)
pick 13134a884 Fix imports in pip3 ubuntu by suffixing affected files (#15891)
pick 4ad4cc632 MNT Raise erorr when normalize is invalid in confusion_matrix (#15888)
d 7c58c2037 [MRG] MNT Adds comment regarding coverage pinning (#15912)
pick b1a336782 [MRG] DOC Increases search results for API object results (#15574)
pick 2d687bfd0 MNT Ignores warning in pyamg for deprecated scipy.random (#15914)
pick a5b67f972 DOC Instructions to troubleshoot Windows path length limit (#15916)
pick cc8d2d26b DOC add versionadded directive to some estimators (#15849)
pick 2e90b8977 DOC clarify doc-string of roc_auc_score and add references (#15293)
pick 88b632287 MNT Adds skip lint to azure pipeline CI (#15904)
pick a6c07f2b0 BLD Fixes bug when building with NO_MATHJAX=1 (#15892)
pick 556eb9246 [MRG] BUG Checks to number of axes in passed in ax more generically (#15760)
pick a4db10152 EXA Minor fixes in plot_sparse_logistic_regression_20newsgroups.py (#15925)
pick 0d3de54c6 BUG Do not shadow public functions with deprecated modules (#15846)
pick 2f86cdb6f Import sklearn._distributor_init first (#15929)
pick 0e10b3a3c DOC Fix typos, via a Levenshtein-style corrector (#15923)
pick 06e9a3fe3 DOC in canned comment, mention that PR title becomes commit me… (#15935)
pick 613a3338f DOC/EXA Correct spelling of "Classification" (#15938)
pick a5542e949 BUG fix pip3 ubuntu update by suffixing file (#15928)
d 9626583e8 ENH Improve column_or_1d error message (#15926)
pick 1b55e2f5c [MRG] Ways to compute center_shift_total were different in "full" and "elkan" algorithms. (#15930)
pick 3a5aced32 TST Fixes integer test for train and test indices (#15941)
pick a68ba9730 BUG ensure that parallel/sequential give the same permutation importances (#15933)
pick fa4646749 Formatting fixes in changelog (#15944)
pick 0c0b6dad6 MRG FIX: order of values of self.quantiles_ in QuantileTransformer (#15751)
pick b42125bec [MRG] BUG Fixes constrast in plot_confusion_matrix (#15936)
pick dfad061da BUG use zero_division argument in classification_report (#15879)
d 0abd48b9c MAINT Unpin coverage package (#15957)
pick f65cdf2d5 DOC change logreg solver in plot_logistic_path (#15927)
pick 791718727 DOC fix whats new ordering (#15961)
d 88963f86f ENH Make cross_val_predict support method="predict_proba" and y=None (#15918)
pick 9408203ac COSMIT use np.iinfo to define the max int32 (#15960)
d e1f7b631c ENH Allow usage of nu=inf in Matern kernel (#15972)
pick aa75e64c8 DOC Apply numpydoc validation to VotingRegressor methods (#15969)
d 1cda8ed98 ENH adding as_frame functionality for CA housing dataset loader (#15950)
pick bbb21e2ac DOC improve naive_bayes.py documentation (#15943)
pick 86b463693 DOC Fix default values in Perceptron documentation (#15965)
pick 8fc16206e DOC Improve default values in logistic documentation (#15966)
pick 2a185f91b DOC Improve documentation of default values for imputers (#15964)
pick c393bdc6d EXA/MAINT Simplify code in manifold learning example (#15949)
d 725ca8f32 MAINT Remove redundant sample_weights check in DummyClassifier (#15510)
pick 3339d8024 DOC Improve default values in SGD documentation (#15967)
d 68c087038 TST Add test for permutation importance with high cardinality (#15939)
pick 74a08746d DOC Improve defaults in neural network documentation (#15968)
pick d163d5ad9 FIX use safe_sparse_dot for callable kernel in LabelSpreading (#15868)
pick 9accce551 BUG Adds attributes back to check_is_fitted (#15947)
pick 2287d2d22 DOC update check_is_fitted what's new
pick d5dcf48c0 DOC change python-devel to python3-devel for yum. (#15986)
pick 14bf5bc25 DOC Correct the default value of values_format in plot_confusion_matrix (#15981)
pick 1e166f77f [MRG] MNT Updates pypy to use 7.2.0 (#15954)
pick dae52f9cb MNT Only tweets on main repo (#15993)
pick eb3ad2dd5 FIX Add missing 'values_format' param to disp.plot() in plot_confusion_matrix (#15937)
pick 46001e7cf FIX support scalar values in fit_params in SearchCV (#15863)
pick 70d8b70d7 Remove abstractmethod that silently brake downstream packages (#15996)
pick f28a90c9a FIX restore BaseNB._check_X without abstractmethod decoration (#15997)
pick 072bfc95c Update v0.22 changelog for 0.22.1 (#16002)
pick c8c21ae18 STY Removes hidden scroll bar (#15999)
```

<del>We also still need #15997.</del> (now included)

cgsavard committed:
DOC fixed default values in dbscan (#15753)

cgsavard committed:
DOC fix incorrect branch reference in contributing doc (#15779)

jnothman committed:
DOC relabel Feature -> Efficiency in change log (#15770)

cgsavard committed:
DOC fixed Birch default value (#15780)

thomasjpfan committed:
STY Minior change on code padding in website theme (#15768)

ogrisel committed:
DOC Fix yticklabels order in permutation importances example (#15799)

* Fix yticklabels order in permutation importances example

* Trigger ci

thomasjpfan committed:
STY Update wrapper width (#15793)

kwinkunks committed:
DOC Long sentence was hard to parse and ambiguous in _classification.py (#15769)

kp1302 committed:
DOC Removed duplicate 'classes_' attribute in Naive Bayes classifiers (#15811)

thomasjpfan committed:
BUG Fixes pandas dataframe bug with boolean dtypes (#15797)

thomasjpfan committed:
BUG Returns only public estimators in all_estimators (#15380)

lucyleeow committed:
DOC improve doc for multiclass and types_of_target (#15333)

thomasjpfan committed:
TST Increases tol for check_pca_float_dtype_preservation assertion (#15775)

JJmistry committed:
update _alpha_grid class in _coordinate_descent.py (#15835)

alfaro96 committed:
FIX Explicit conversion of ndarray to object dtype. (#15832)

thomasjpfan committed:
BLD Parallelize sphinx builds on circle ci (#15745)

SylvainLan committed:
DOC correct url for preprocessing (#15853)

NicolasHug committed:
MNT avoid generating too many cross links in examples (#15844)

qinhanmin2014 committed:
DOC Correct wrong doc in precision_recall_fscore_support (#15833)

adrinjalali committed:
DOC add comment in check_pca_float_dtype_preservation (#15819)

Documenting the changes in https://github.com/scikit-learn/scikit-learn/pull/15775

qinhanmin2014 committed:
DOC correct indents in docstring _split.py (#15843)

cgsavard committed:
DOC fix docstring of KMeans based on sklearn guideline (#15754)

vachanda committed:
DOC fix docstring of AgglomerativeClustering based on sklearn guideline (#15764)

cgsavard committed:
DOC fix docstring of AffinityPropagation based on sklearn guideline (#15777)

cgsavard committed:
DOC fixed SpectralCoclustering and SpectralBiclustering docstrings following sklearn guideline (#15778)

vachanda committed:
DOC fix FeatureAgglomeration and MiniBatchKMeans docstring following sklearn guideline (#15809)

qinhanmin2014 committed:
TST Specify random_state in test_cv_iterable_wrapper (#15829)

sambhav committed:
DOC Include LinearSV{C, R} in models that support sample_weights (#15871)

qinhanmin2014 committed:
DOC correct some indents (#15875)

alfaro96 committed:
DOC Fix documentation of default values in tree classes (#15870)

wenliwyan committed:
DOC fix typo in docstring (#15887)

shivamgargsya committed:
DOC FIX default value for xticks_rotation in plot_confusion_matrix (#15890)

thomasjpfan committed:
Fix imports in pip3 ubuntu by suffixing affected files (#15891)

qinhanmin2014 committed:
MNT Raise erorr when normalize is invalid in confusion_matrix (#15888)

thomasjpfan committed:
[MRG] DOC Increases search results for API object results (#15574)

thomasjpfan committed:
MNT Ignores warning in pyamg for deprecated scipy.random (#15914)

ogrisel committed:
DOC Instructions to troubleshoot Windows path length limit (#15916)

Commit:
DOC add versionadded directive to some estimators (#15849)

oulenz committed:
DOC clarify doc-string of roc_auc_score and add references (#15293)

thomasjpfan committed:
MNT Adds skip lint to azure pipeline CI (#15904)

thomasjpfan committed:
BLD Fixes bug when building with NO_MATHJAX=1 (#15892)

thomasjpfan committed:
[MRG] BUG Checks to number of axes in passed in ax more generically (#15760)

lesteve committed:
EXA Minor fixes in plot_sparse_logistic_regression_20newsgroups.py (#15925)

ogrisel committed:
BUG Do not shadow public functions with deprecated modules (#15846)

ogrisel committed:
Import sklearn._distributor_init first (#15929)

bwignall committed:
DOC Fix typos, via a Levenshtein-style corrector (#15923)

jnothman committed:
DOC in canned comment, mention that PR title becomes commit me… (#15935)

ritchieng committed:
DOC/EXA Correct spelling of "Classification" (#15938)

glemaitre committed:
BUG fix pip3 ubuntu update by suffixing file (#15928)

inder128 committed:
[MRG] Ways to compute center_shift_total were different in "full" and "elkan" algorithms. (#15930)

scibol committed:
TST Fixes integer test for train and test indices (#15941)

glemaitre committed:
BUG ensure that parallel/sequential give the same permutation importances (#15933)

ogrisel committed:
Formatting fixes in changelog (#15944)

tirthasheshpatel committed:
MRG FIX: order of values of self.quantiles_ in QuantileTransformer (#15751)

thomasjpfan committed:
[MRG] BUG Fixes constrast in plot_confusion_matrix (#15936)

Commit:
BUG use zero_division argument in classification_report (#15879)

agramfort committed:
DOC change logreg solver in plot_logistic_path (#15927)

glemaitre committed:
DOC fix whats new ordering (#15961)

glemaitre committed:
COSMIT use np.iinfo to define the max int32 (#15960)

Commit:
DOC Apply numpydoc validation to VotingRegressor methods (#15969)

Co-authored-by: Tiffany R. Williams <Tiffany8@users.noreply.github.com>

Commit:
DOC improve naive_bayes.py documentation (#15943)

Co-authored-by: Jigna Panchal <40188288+jigna-panchal@users.noreply.github.com>

ankishb committed:
DOC Fix default values in Perceptron documentation (#15965)

ankishb committed:
DOC Improve default values in logistic documentation (#15966)

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>

pulkitmehtawork committed:
DOC Improve documentation of default values for imputers (#15964)

DavidBreuer committed:
EXA/MAINT Simplify code in manifold learning example (#15949)

ankishb committed:
DOC Improve default values in SGD documentation (#15967)

ankishb committed:
DOC Improve defaults in neural network documentation (#15968)

nik-sm committed:
FIX use safe_sparse_dot for callable kernel in LabelSpreading (#15868)

thomasjpfan committed:
BUG Adds attributes back to check_is_fitted (#15947)

Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>

qinhanmin2014 committed:
DOC update check_is_fitted what's new

Windber committed:
DOC change python-devel to python3-devel for yum. (#15986)

qinhanmin2014 committed:
DOC Correct the default value of values_format in plot_confusion_matrix (#15981)

thomasjpfan committed:
[MRG] MNT Updates pypy to use 7.2.0 (#15954)

blynotes committed:
FIX Add missing 'values_format' param to disp.plot() in plot_confusion_matrix (#15937)

adrinjalali committed:
FIX support scalar values in fit_params in SearchCV (#15863)

* support a scalar fit param

* pep8

* TST add test for desired behavior

* FIX introduce _check_fit_params to validate parameters

* DOC update whats new

* TST tests both grid-search and randomize-search

* PEP8

* DOC revert unecessary change

* TST add test for _check_fit_params

* olivier comments

* TST fixes

* DOC whats new

* DOC whats new

* TST revert type of error

* add olivier suggestions

* address olivier comments

* address thomas comments

* PEP8

* comments olivier

* TST fix test by passing X

* avoid to call twice tocsr

* add case column/row sparse in check_fit_param

* provide optional indices

* TST check content when indexing params

* PEP8

* TST update tests to check identity

* stupid fix

* use a distribution in RandomizedSearchCV

* MNT add lightgbm to one of the CI build

* move to another build

* do not install dependencies lightgbm

* MNT comments on the CI setup

* address some comments

* Test fit_params compat without dependency on lightgbm

Co-authored-by: Guillaume Lemaitre <g.lemaitre58@gmail.com>
Co-authored-by: Olivier Grisel <olivier.grisel@ensta.org>

bsipocz committed:
Remove abstractmethod that silently brake downstream packages (#15996)

ogrisel committed:
FIX restore BaseNB._check_X without abstractmethod decoration (#15997)

jnothman commented on doc/whats_new/v0.22.rst:
```diff
@@ -2,6 +2,100 @@
+Version 0.22.1
+==============
+
+**In Development**
```
this needs updating here and in master

jnothman commented on doc/whats_new/v0.22.rst:
```diff
@@ -764,6 +858,10 @@ Changelog
   :class:`preprocessing.KernelCenterer`
   :pr:`14336` by :user:`Gregory Dexter <gdex1>`.
 
+- |Fix| :class:`preprocessing.QuantileTransformer` now guarantees the 
```
This should be under 0.22.1 here and in master

jnothman commented:
Commits I might have thought to include:
```
d cc88ec97b DOC fixed default values in dbscan (#15753)
d 0bbe923fe DOC fixed Birch default value (#15780)
d 5e66d8ee2 STY Update wrapper width (#15793)
```

There are currently (severe) flake8 issues:
```
Running flake8 on the diff in the range db6c12fc1..d5f70c190 (148 commit(s)):
--------------------------------------------------------------------------------
sklearn/cluster/_agglomerative.py:772:6: F821 undefined name 'deprecated'
    @deprecated("The ``n_components_`` attribute was deprecated "
     ^
sklearn/metrics/_plot/tests/test_plot_confusion_matrix.py:232:1: E303 too many blank lines (4)
@pytest.mark.parametrize(
^
```

I can't see this first issue in the diff, though...

jnothman reviewed (commented):
Thanks for taking this on, @ogrisel! I'm glad this is going to be released, and I've been busier than expected...

`__version__` in `sklearn/__init__.py` also needs updating

ogrisel commented:
Yes there are some errors, I am on it, sorry for the delay.

jnothman commented:
No worries :)
Happy new year!

ogrisel committed:
Update v0.22 changelog for 0.22.1 (#16002)

- set the date
- move entry for quantile transformer to the 0.22.1 section
- fix alphabetical ordering of modules

ogrisel commented:
The changelog has been fixed in #16002. Will redo the rebase and include the missing element along with #15997.

ogrisel commented:
Commits I might have thought to include:
```
d 5e66d8ee2 STY Update wrapper width (#15793)
```

I did not want to backport this because of https://github.com/scikit-learn/scikit-learn/pull/15793#issuecomment-569961898. There is now an alternative for master being worked on in #15999.

thomasjpfan committed:
STY Removes hidden scroll bar (#15999)

ogrisel commented:
Ok I did the new rebase but I still need to do some manual updates on top of that to fix linting, missing deprecated import and versions (and maybe other things.

ogrisel committed:
Flake8 fixes

ogrisel committed:
Fix: remove left-over lines that should have been deleted during conflict resolution when rebasing

ogrisel committed:
Fix missing imports

ogrisel committed:
Update version

ogrisel committed:
Fix test_check_is_fitted

adrinjalali commented:
Wow, that's a massive diff for a minor release. Thanks @ogrisel . 

Should we include those linter issues in master?

Also, shouldn't we include the contributors' names to the changelog?

ogrisel committed:
Make test_sag_regressor_computed_correctly deterministic (#16003)

Fix #15818.

ogrisel commented:
> Should we include those linter issues in master?

I believe most of the linting issues where caused by the backporting action it-self.

> Also, shouldn't we include the contributors' names to the changelog?

We usually do not do it for minor releases.

ogrisel commented:
Ok, let's merge and tag.
